src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/pipeline/JiraSendBuildInfoStep.java

```diff
@@ -100,9 +106,16 @@ public JiraSendBuildInfoStepExecution(
         @Override
         protected JiraSendInfoResponse run() throws Exception {
             final TaskListener taskListener = getContext().get(TaskListener.class);
-            final WorkflowRun build = getContext().get(WorkflowRun.class);
+            final WorkflowRun workflowRun = getContext().get(WorkflowRun.class);
+            final Optional<String> site = pickDefaultSite(step.getSite());
+
+            if (!site.isPresent()) {
+                workflowRun.setResult(Result.FAILURE);
+                getContext().setResult(Result.FAILURE);
+                throw new DefaultSiteLookupFailureException();
```

Comment:
I don't think we should fail a build if no default site can be looked up. This is inconsistent with the current plugin behaviour. For example, if no Jira sites are configured, we just log a warning in the Output console, but we don't fail a build.